### ISSUE TITLE
[0.9.0] Setting data & status on root span has no impact

### ISSUE THREAD

**Issue by Mokto** [getsentry/sentry-go#311]
## Summary
Hi, i've been using the new tracing system from the golang library, good job!
I've been unable to set data & level on a trace, though

## Steps To Reproduce

```go
span := sentry.StartSpan(ctx, name)
span.Status = sentry.SpanStatusInternalError

span.Data = map[string]interface{}{
"body":        c.Request().Body,
"path":        c.Path(),
"queryParams": c.QueryParams(),
"paramValues": c.ParamValues(),
}
span.Finish()
```

## Expected Behavior
See the data in the UI

### SDK
 - `sentry-go` version: 0.9.0
 - Go version: 1.15.0
 - Using Go Modules? yes

### Sentry
 - Using hosted Sentry in sentry.io? no
 - Using your own Sentry installation? Version: 20.12.1

**Comment by rhcarvalho** [comment]
@Mokto could you confirm that the values you set on Data are JSON-serializable?
To clarify, do you see the transaction event in the UI but without those fields? You can compare by Event ID by setting Debug to true on sentry.Init.

**Comment by Mokto** [comment]
Hi,

1) Yes I tried with :
```
span.Data = map[string]interface{}{
"test": "a",
}
```

2) Yes I do see the transaction, also child spans are working if I'm using any. + reported by debug: true
3) It's worth noting that when I set a status on a span, the status is not visible on both the transaction & the span. Status reports empty
4) About adding some data. I don't see any in the UI, but I'm not sure where to look to be honest!

**Comment by Mokto** [comment]
Ok, I think I have more insight!

I am able to set data & status on any child span, but not on the root span.

![sentry](https://user-images.githubusercontent.com/5103928/102464968-5ce36d00-404d-11eb-806c-d7c91fd6dcd5.png)

The weird thing here is that I'm naming my parent span `http` but in the UI it always shows the name of the first child span.

Maybe I'm doing something wrong here ?

Thanks!

**Comment by rhcarvalho** [comment]
I'm on mobile only atm, will ask someone for a screenshot of where transaction/span status and data show up. The transaction status is on the right hand side of the screen in Sentry, and the data is towards the bottom of the screen, often requiring a scroll. Individual span info is upon clicking the span as your screenshot above.

**Comment by Mokto** [comment]
I just tried it tracing with NodeJS, and everything works fine, I can see the set status & data on the root span.
I changed my description, the issue is only present on the root span.

And once again, weirdly the name of the root span is replaced by the name of one of the child.

In the NodeJS example they're using `startTransaction`, which might do something else than `startSpan` ? Just a wild (and probably wrong) guess.
